e2e/tests-dfx/assetscanister.bash

```diff
@@ -14,6 +14,34 @@ teardown() {
     standard_teardown
 }
 
+@test "authorize and deauthorize work as expected" {
+  PRINCIPAL=$(dfx identity get-principal)
+  install_asset assetscanister
+  dfx_start
+  assert_command dfx deploy
+  [ "$USE_IC_REF" ] && dfx canister update-settings --add-controller "$(dfx canister id e2e_project_frontend)" --all # Dec. 2022: ic-ref has not received the change yet that a canister is always allowed to check it's status
+
+  # deployer is automatically authorized
+  assert_command dfx canister call e2e_project_frontend list_authorized '()'
+  assert_contains "$PRINCIPAL"
+
+  assert_command dfx canister call e2e_project_frontend deauthorize "(principal \"$PRINCIPAL\")"
```

Comment:
Let's have this test demonstrate that `deauthorize` can only be called by an authorized caller